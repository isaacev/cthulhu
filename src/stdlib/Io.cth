intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
