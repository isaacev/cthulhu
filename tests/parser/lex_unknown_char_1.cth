use ::Io;

fn main() -> () {
  let x = %123;
};
