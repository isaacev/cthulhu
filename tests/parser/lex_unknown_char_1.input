use ::IO;

fn main() -> () {
  let x = %123;
};
