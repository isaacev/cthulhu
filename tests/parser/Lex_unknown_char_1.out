
  EXPECTED EXPRESSION  -------------------------------------------------------

    : Lex_unknown_char_1
  4 |   let x = %123;
    |           ^

  An expression can be like one of the following:

    a + b * c

    myFunction("hello")

    if a { b; } else { c; }

